File: Backend/TruckEase/TruckEase/Contracts/TransportOffer/Mappers/TransportOfferMapper.cs
namespace TruckEase.Contracts.TransportOffer.Mappers;

using TruckEase.Contracts.TransportOffer.Requests;
using TruckEase.Contracts.TransportOffer.Responses;
using TruckEase.Dtos;
using TruckEase.Vtos;

public static class TransportOfferMapper
{
    public static CreateTransportOfferVto ToCreateTransportVto(this TransportOfferRequest request)
    {
        return new CreateTransportOfferVto
        {
            AdditionalInfo = request.AdditionalInfo,
            Price = request.Price
        };
    }

    public static OfferInfoResponse ToTransportOfferInfoResponse(this OfferInfoDto requestDto)
    {
        return new OfferInfoResponse
        {
            OfferId = requestDto.OfferId,
            AdditionalInfo = requestDto.AdditionalInfo,
            Price = requestDto.Price,
            OfferStatus = requestDto.OfferStatus,
            StartTime = requestDto.StartTime,
            StartLocation = requestDto.StartLocation,
            ArrivalTime = requestDto.ArrivalTime,
            Destination = requestDto.Destination,
            TransportType = requestDto.TransportType,
            RequesterCompanyId = requestDto.RequesterCompanyId

        };
    }


}

File: Frontend/TruckEase/TruckEaseApp/src/app/models/mappers/transport-offer.mapper.ts
import { TransportOfferInfoResponse } from "../responses/transport-offer-info.response";
import { TransportOfferInfoViewModel } from "../view-models/transport-offer-info.view-model";

export namespace TransportOfferMapper {
    export function toTransportOfferViewModel(response: TransportOfferInfoResponse): TransportOfferInfoViewModel {
  
      return {
        offerId: response.offerId,
        additionalInfo: response.additionalInfo,
        price: response.price,
        offerStatus: response.offerStatus,
        startTime: response.startTime,
        arrivalTime: response.arrivalTime,
        startLocation: response.startLocation,
        destination: response.destination,
        transportType: response.transportType,
        requesterCompanyId: response.requesterCompanyId
    }
  
  }
  }
  